src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -17,7 +17,11 @@ public Task Complete(IContainSagaData sagaData, SynchronizedStorageSession sessi
             inMemSession.Enlist(() =>
             {
                 VersionedSagaEntity value;
-                data.TryRemove(sagaData.Id, out value);
+                if (data.TryRemove(sagaData.Id, out value))
+                {
+                    object lockToken;
+                    lockers.TryRemove(value.LockTokenKey, out lockToken);
```

Comment:
Since we are only ever removing here and never modifying I think we should be safe with a `TryRemove`